generator/generator_tests/altitude_test.cpp

```diff
@@ -0,0 +1,216 @@
+#include "testing/testing.hpp"
+
+#include "generator/altitude_generator.hpp"
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "routing/road_graph.hpp"
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/altitude_loader.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/index.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/platform.hpp"
+
+#include "platform/platform_tests_support/scoped_dir.hpp"
+#include "platform/platform_tests_support/scoped_file.hpp"
+#include "platform/platform_tests_support/writable_dir_changer.hpp"
+
+#include "base/logging.hpp"
+#include "base/scope_guard.hpp"
+
+#include "std/string.hpp"
+
+using namespace feature;
+using namespace generator;
+using namespace platform;
+using namespace routing;
+
+namespace
+{
+// These tests generate mwms with altitude sections and then check if altitudes
+// in the mwms are correct. The mwms are initialized with different sets of features.
+// There are several restrictions for these features:
+// * all of them are linear
+// * all coords of these features should be integer (it's necessary for easy implementation of MockAltitudeGetter)
+// * if accoding to one feature a point has some altitude, the point of another feature
+//   with the same coords has to have the same altitude
+// @TODO(bykoianko) Add ability to add to the tests not road features without altitude information.
+
+// Directory name for creating test mwm and temporary files.
+string const kTestDir = "altitude_generation_test";
+// Temporary mwm name for testing.
+string const kTestMwm = "test";
+
+struct Point3D
+{
+  Point3D(int32_t x, int32_t y, TAltitude a) : m_point(x, y), m_altitude(a) {}
+
+  m2::PointI m_point;
+  TAltitude m_altitude;
+};
+
+using TPoint3DList = vector<Point3D>;
+
+TPoint3DList const kRoad1 = {{0, -1, -1}, {0, 0, 0}, {0, 1, 1}};
+TPoint3DList const kRoad2 = {{0, 1, 1}, {5, 1, 1}, {10, 1, 1}};
+TPoint3DList const kRoad3 = {{10, 1, 1}, {15, 6, 100}, {20, 11, 110}};
+TPoint3DList const kRoad4 = {{-10, 1, -1}, {-20, 6, -100}, {-20, -11, -110}};
+
+class MockAltitudeGetter : public IAltitudeGetter
+{
+public:
+  using TMockAltitudes = map<m2::PointI, TAltitude>;
```

Comment:
In MockAltitudeGetter::GetAltitude(m2::PointD const & p) we need to find altitude by 2D point. So I need to find a point in 2D array by its coors. I have a choice of brute force with almost equal, KD-tree and "rounding".